client/my-sites/email/email-providers-stacked-comparison/email-provider-stacked-card/email-provider-generic-form.tsx

```diff
@@ -0,0 +1,868 @@
+	value,
+	error:
+		! error && 60 < value.length
+			? translate( "This field can't be longer than %s characters.", {
+					args: '60',
+			  } )
+			: error,
+} );
+
+/*
+ * Adds a new error if the specified mailbox field has no errors and appears more than once in the provided map.
+ */
+const validateNewUserMailboxIsUnique = (
+	{ value: mailBox, error: previousError }: GenericNewUserField,
+	mailboxesByCount: { [ mailbox: string ]: number }
+) => ( {
+	value: mailBox,
+	error:
+		mailboxesByCount[ mailBox ] > 1
+			? translate( 'Please use a unique mailbox for each user.' )
+			: previousError,
+} );
+
+/*
+ * Adds a duplicate error to each mailBox with a duplicate mailbox
+ */
+const validateNewUsersAreUnique = ( users: GenericNewUser[] ): GenericNewUser[] => {
+	const mailboxesByCount: { [ mailbox: string ]: number } = countBy(
+		users.map( ( { mailBox: { value: mailBox } } ) => mailBox )
+	);
+
+	return users.map(
+		( {
+			uuid,
+			alternativeEmail,
+			fullName,
+			email,
+			domain,
+			isAdmin,
+			mailBox,
+			firstName,
+			lastName,
+			password,
+		} ) => ( {
+			uuid,
+			alternativeEmail,
+			email,
+			firstName,
+			fullName,
+			lastName,
+			domain,
+			isAdmin,
+			mailBox: validateNewUserMailboxIsUnique( mailBox, mailboxesByCount ),
+			password,
+		} )
+	);
+};
+
+/*
+ * Validates the specified password field.
+ *
+ * @see https://support.google.com/accounts/answer/32040 for requirements
+ */
+export const validatePasswordField = (
+	{ value, error }: GenericNewUserField,
+	minimumLength = 12
+): GenericNewUserField => {
+	if ( ! error && minimumLength > value.length ) {
+		return {
+			value,
+			error: translate( "This field can't be shorter than %s characters.", {
+				args: String( minimumLength ),
+			} ),
+		};
+	}
+
+	if ( ! error && 100 < value.length ) {
+		return {
+			value,
+			error: translate( "This field can't be longer than %s characters.", { args: '100' } ),
+		};
+	}
+
+	if ( ! error && value.startsWith( ' ' ) ) {
+		return {
+			value,
+			error: translate( "This field can't start with a white space." ),
+		};
+	}
+
+	// Checks that passwords only have ASCII characters (see https://en.wikipedia.org/wiki/ASCII#Character_set)
+	const regexp = /[^\x20-\x7E]/;
+
+	if ( ! error && regexp.test( value ) ) {
+		const firstForbiddenCharacter = [ ...value ].find( ( character ) => regexp.test( character ) );
+
+		return {
+			value,
+			error: translate( "This field can't accept '%s' as character.", {
+				args: firstForbiddenCharacter,
+				comment: '%s denotes a single character that is not allowed in this field',
+			} ),
+		};
+	}
+
+	if ( ! error && value.endsWith( ' ' ) ) {
+		return {
+			value,
+			error: translate( "This field can't end with a white space." ),
+		};
+	}
+
+	return { value, error };
+};
+
+/*
+ * Run all validations on a user.
+ */
+const validateUser = ( user: GenericNewUser, optoinalFields: string[] ): GenericNewUser => {
```

Comment:
Typo: `optoinalFields` => `optionalFields`
```suggestion
const validateUser = ( user: GenericNewUser, optionalFields: string[] ): GenericNewUser => {
```